cmd/github-ldap-user-group-creator/main.go

```diff
@@ -290,6 +295,49 @@ type GroupClusters struct {
 
 var githubRobotIds = sets.New[string]("RH-Cachito", "openshift-bot", "openshift-ci-robot", "openshift-merge-robot")
 
+func deleteUsersNotInRover(ctx context.Context, clients map[string]ctrlruntimeclient.Client,
+	mapping map[string]string, dryRun bool) error {
+
+	var errs []error
+	roverUsers := sets.New[string]()
+	for _, kerberosId := range mapping {
+		sets.Insert[string](roverUsers, kerberosId)
+	}
+
+	for cluster, client := range clients {
+		usersNotInRover, _ := getUsersNotInRover(ctx, client, cluster, roverUsers)
+		logrus.Infoln("Deleting users on cluster ", cluster)
+		if dryRun {
+			logrus.Infoln("Dry run, not deleting users")
+			continue
+		}
+		logrus.Infoln(usersNotInRover)
+		//DELETE USERS + IDENTITIES
+	}
+
+	return kerrors.NewAggregate(errs)
+}
+
+func getUsersNotInRover(ctx context.Context, client ctrlruntimeclient.Client,
+	cluster string, roverUsers sets.Set[string]) (sets.Set[string], error) {
+
+	var errs []error
+
+	identities := &userv1.IdentityList{}
+	if err := client.List(ctx, identities); err != nil {
+		errs = append(errs, fmt.Errorf("failed to list identities on cluster %s: %w", cluster, err))
+	}
+	identitiesOnCluster := sets.New[string]()
```

Comment:
- list users on the cluster
- loop on the users to collect the ones without `kerberosIDs.has(user.metadata.Name)`
- return the list of those users

In the caller:
loop over the list to delete those users and all its identities (`user.identities`).